File: render.rb
# frozen_string_literal: true

require 'toml-rb'
require 'erb'


class Config
    attr_reader :data

    def initialize(sourcefile)
        @data = TomlRB.load_file(sourcefile)
    end

    def find_or_blank(keys)
        if keys.class.method_defined? :each
            begin
                @data.dig *keys
            rescue
                nil
            end
        if @data[key]

"<div class="row"><div>Project Name and Location: </div><% if project['name'] %><div style="text-decoration:underline"><%= project['name'] %></div><% else %><div class="underline"></div><% end %></div>
"

    def b
        binding
    end
end

config = Config.new('project.toml')

template = File.read('templates/SWPPP_template.erb')
html = ERB.new(template).result config.b

File.open('templates/check.html', 'w') { |f| f.write(html) }

puts 'done'

File: render.py
import sys
from datetime import datetime
from jinja2.loaders import FileSystemLoader
from typing import Dict, Any, List
from pdb import set_trace

import toml
from jinja2 import Environment


def usage():
    print("""python render.py FILENAME""")
    quit()


def content_or_blank(local_config:Dict[str, Any], keys:List[str]) -> str:
    """Keys must be passed in list format in order of most to least specific.
    ie) {key1:{key2}} => [key2,key1]
    :type local_config: Dict[str,Any]
    :type keys: List[str]
    :rtype str
    """
    search_key = keys.pop()
    result = local_config.get(search_key)
    if result and keys:
        return content_or_blank(result, keys)
    if result:
        return f"<div class='underlined'>{result}</div>"
    else:
        return "<div class='blank-line'></div>"


if __name__ == "__main__":
    if len(sys.argv) < 1:
        usage()
    with open(sys.argv[1], 'r') as config_file:
        config = toml.load(config_file)
        config['today'] = datetime.now().strftime("%B %d, %Y")
    env = Environment(loader=FileSystemLoader('templates'))
    env.filters['content_or_blank'] = content_or_blank
    template = env.get_template('SWPPP_template.html')
    with open('check.html', 'w') as out_file:
        out_file.write(template.render(config))